scipp_widgets/widgets.py

```diff
@@ -0,0 +1,150 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+# @file
+# @author Matthew Andrew
+
+import ipywidgets as widgets
+from .inputs import get_notebook_global_scope
+from IPython.core.display import display, Javascript
+from typing import Callable
+
+javascript_functions = {False: "hide()", True: "show()"}
+
+
+def toggle_code(state, output_widget=None):
+    """
+    Toggles the JavaScript show()/hide() function on the div.input element.
+    """
+    output_string = "this.element.closest('.cell').children('.input').{}"
+    output_args = (javascript_functions[state], )
+    output = output_string.format(*output_args)
+
+    if (output_widget):
+        output_widget.clear_output()
+        with output_widget:
+            display(Javascript(output))
+    else:
+        display(Javascript(output))
+
+
+class HideCodeWidget(widgets.Box):
+    def __init__(self, state):
+        super().__init__()
+        self.output_widget = widgets.Output()
+        self.button = widgets.ToggleButton(not state, description='Py')
+        self.button.observe(self._on_button_toggle, "value")
+        self.button.layout.flex = '0 1 40px'
+        self.button.layout.align_self = 'flex-end'
+
+        self.children = [self.button, self.output_widget]
+        self.layout.align_self = 'flex-end'
+        toggle_code(not state)
+
+    def _on_button_toggle(self, value):
+        toggle_code(value.new, self.output_widget)
+
+
+class DisplayWidget(widgets.Box):
+    """
+    Provides a simple graphical wrapper around a given callable.
+    """
+    def __init__(self, callab, name: str, inputs, hide_code=False):
+        """
+        Parameters:
+        callable (Callable): The function to call
+        name: name of widget to display
+        inputs (Input object): class containing input data
+        """
+        super().__init__()
+        self.layout.flex_flow = 'column'
+        self.callable = callable
```

Comment:
```suggestion
        self.callable = callab
```

Reply:
This was he cause of your issue above. Have now pushed a fix.